frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue

```diff
@@ -227,8 +406,213 @@ export default class CaseEdit extends Vue {
     { type: T.problemCreatorLineMatrix, kind: 'matrix' },
   ];
 
+  matrixDistinctOptions: {
+    type: string;
+    distinctType: MatrixDistinctType;
+  }[] = [
+    { type: T.matrixModalDistinctNone, distinctType: 'none' },
+    { type: T.matrixModalDistinctRow, distinctType: 'rows' },
+    { type: T.matrixModalDistinctColumn, distinctType: 'cols' },
+    { type: T.matrixModalDistinctAll, distinctType: 'both' },
+  ];
+
+  editModalState(kind: CaseLineKind) {
+    if (kind === 'array') {
+      this.matrixModalEdit = false;
+      this.arrayModalEdit = true;
+    } else if (kind === 'matrix') {
+      this.arrayModalEdit = false;
+      this.matrixModalEdit = true;
+    }
+  }
+
   getLineNameFromKind(kind: CaseLineKind) {
-    return this.lineOptions.find((line) => line.kind === kind)?.type;
+    return this.lineKindOptions.find((row) => row.kind === kind)?.type;
+  }
+
+  getDistinctNameFromType(distinctype: MatrixDistinctType) {
+    return this.matrixDistinctOptions.find(
+      (row) => row.distinctType === distinctype,
+    )?.type;
+  }
+
+  getDistinctArrayContents(
+    size: number,
+    low: number = 0,
+    high: number = 0,
+  ): string {
+    const generatedArray = new Set<number>();
+    while (generatedArray.size < size) {
+      generatedArray.add(
+        Number(low.toString()) + Math.floor(Math.random() * (high - low + 1)),
```

Comment:
Is it necessary to do the double parsing, first to a string, and then to a number? If the original type of this attribute is already a number, can we eliminate that double parsing?

```suggestion
        low + Math.floor(Math.random() * (high - low + 1)),
```

The same applies to the other places where you use it.

Reply:
Hi @pabo99, double-parsing looks a bit weird, but when I was not using it, the typescript function was doing well, but the modal-component was giving something else, I can probably reproduce this and will share here.

Reply:
Weird, I cannot reproduce, as far as I remember, the add function was concatenating the two numbers instead of adding even when typed statically and strongly,

I did something like this:
```ts
while (generatedArray.size < size) {
    generatedArray.add(Math.floor(Math.random() * (high - low + 1)) + low);
}
```

Reply:
It could be, but normally this should only perform arithmetic operations, as all the variables are numeric. 

If we find the issue once you push your changes, we will try to fix it.